crates/red_knot_python_semantic/resources/mdtest/declaration/with.md

```diff
@@ -0,0 +1,122 @@
+# With declarations
```

Comment:
Nit about this title, and the location of this file: with statements create Definitions, which are Bindings, but are not Declarations. (A declaration is only an annotated assignment, or a function parameter, or a function or class definition.) So I think this title should be just `With statements`.

Regarding file location, we don't have a general directory for all kinds of definitions/bindings, so I think `with/` could be a top-level directory if/when we have multiple files to put in there; for now a top level `with.md` would be fine, I think.